src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
 public class WorkWithFile {
+    private ArrayList<DataFromFile> dataList = new ArrayList<>();
```

Comment:
this is service class, so it should not store any state
let's refactor your code so store data in local variables and pass them from method to method